ast/builtins.go

```diff
@@ -2649,6 +2655,73 @@ var GraphQLSchemaIsValid = &Builtin{
 	),
 }
 
+/**
+ * JSON Schema
+ */
+
+// JSONSchemaIsValid returns true if the input is valid JSON schema,
+// and returns false for all other inputs.
+var JSONSchemaIsValid = &Builtin{
+	Name:        "jsonschema.is_valid",
+	Description: "Checks that the input is a valid JSON schema object. The schema can be either a JSON string or an JSON object.",
+	Decl: types.NewFunction(
+		types.Args(
+			types.Named("schema", types.NewAny(types.S, types.NewObject(nil, types.NewDynamicProperty(types.A, types.A)))),
+		),
+		types.Named("output", types.B).Description("`true` if the schema is a valid JSON schema. `false` otherwise."),
+	),
+}
+
+// JSONSchemaValidate returns empty string if the input is valid JSON schema
+// and returns error string for all other inputs.
+var JSONSchemaValidate = &Builtin{
+	Name:        "jsonschema.validate",
+	Description: "Checks that the input is a valid JSON schema object. The schema can be either a JSON string or an JSON object.",
+	Decl: types.NewFunction(
+		types.Args(
+			types.Named("schema", types.NewAny(types.S, types.NewObject(nil, types.NewDynamicProperty(types.A, types.A)))),
+		),
+		types.Named("output", types.S).Description("empty string if the schema is a valid JSON schema. Error string otherwise."),
+	),
+}
+
+// JSONIsMatchSchema returns true if the document matches the JSON schema, otherwise false.
+var JSONIsMatchSchema = &Builtin{
+	Name:        "json.is_match_schema",
+	Description: "Checks that the document matches the JSON schema.",
+	Decl: types.NewFunction(
+		types.Args(
+			types.Named("document", types.NewAny(types.S, types.NewObject(nil, types.NewDynamicProperty(types.A, types.A)))),
+			types.Named("schema", types.NewAny(types.S, types.NewObject(nil, types.NewDynamicProperty(types.A, types.A)))),
+		),
+		types.Named("output", types.B).Description("`true` if the document matches the JSON schema. Otherwise `false`."),
+	),
+}
+
+// JSONMatchSchema returns empty array if the document matches the JSON schema,
```

Comment:
Do we need both `json.is_match_schema` and `json.match_schema` then? It seems like this rego snippet would do to implement the former:
```rego
is_match_schema(doc, schema) if json.match_schema(doc, schema) == []
```

Reply:
Agree. Do we also need `jsonschema.is_valid`?
This can be implemented in the same way using `jsonschema.verify`.